### PR TITLE
Make closing the gist flyout more reliable

### PR DESCRIPTION
**Pull request by aomarks**
- You can now close the flyout by clicking the "Share" button again. Previously this didn't work because clicking on it 1. caused the flyout to close itself (because it closes itself when it detects a click outside of itself), and then 2. the share button to open the flyout again (because it naively opened for every click).

- You can now close the flyout by clicking on the preview. Previously this didn't work because those clicks went to the iframe window, and never reached the host window. Now we temporarily set `pointer-events: none` when the flyout is open so that the main window gets those events instead.

### CODE REVIEW

File: packages/lit-dev-content/src/components/litdev-playground-share-button.ts [hunk 1]
```diff
@@ -115,8 +115,13 @@ export class LitDevPlaygroundShareButton extends LitElement {
   }
 
   override update(changes: PropertyValues) {
-    if (changes.has('_open') && this._open) {
-      this._longUrl?.generateUrl();
+    if (changes.has('_open')) {
+      if (this._open) {
+        this._longUrl?.generateUrl();
+        this.dispatchEvent(new Event('opened'));
```
Comment (AndrewJakubowicz) [inline packages/lit-dev-content/src/components/litdev-playground-share-button.ts:126]: Question: Worth registering event types into the TypeScript event map or making it more specific to the share button?

**Review by AndrewJakubowicz — APPROVED** [review]
LGTM and I manually tested in Safari and Chrome.

The iframe handling is genius.

File: packages/lit-dev-content/src/components/litdev-playground-share-button.ts (hunk 1 above)
Comment (aomarks) [reply packages/lit-dev-content/src/components/litdev-playground-share-button.ts:126]: There's no data in this event, and the event map doesn't inform any type checking or anything, so I guess it wouldn't achieve anything practical.

Also `'opened'` is such a generic name that it is likely to collide, and there's no way to declare an event tied to a specific element, right?

Comment (aomarks) [reply packages/lit-dev-content/src/components/litdev-playground-share-button.ts:126]: That's true, maybe prefixing things with the CE name would be a good convention... haven't been doing that generally in this code base, but something to consider as a code improvement generally?

**Pull request merged**